nexus/db-queries/src/db/datastore/mod.rs

```diff
@@ -262,25 +268,95 @@ impl DataStore {
             || async {
                 if let Some(try_for) = try_for {
                     if std::time::Instant::now() > start + try_for {
-                        return Err(BackoffError::permanent(()));
+                        return Err(BackoffError::permanent(
+                            "Timeout waiting for DataStore::new_with_timeout",
+                        ));
                     }
                 }
 
-                match datastore
-                    .ensure_schema(&log, EXPECTED_VERSION, config)
-                    .await
-                {
-                    Ok(()) => return Ok(()),
-                    Err(e) => {
-                        warn!(log, "Failed to ensure schema version"; "error" => #%e);
+                loop {
+                    let checked_action = datastore
+                        .check_schema_and_access(
+                            identity_check,
+                            EXPECTED_VERSION,
+                        )
+                        .await
+                        .map_err(|err| {
+                            warn!(
+                                log,
+                                "Cannot check schema version / Nexus access";
+                                "error" => InlineErrorChain::new(err.as_ref()),
```

Comment:
Tiny nit - `InlineErrorChain` implements `slog::KV` and provides the `"error"` key, so this can be shortened to

```suggestion
                                InlineErrorChain::new(err.as_ref()),
```

(Same nit below, but also I don't feel strongly about this so if you prefer the explicit key, feel free to leave them.)

Reply:
Patched in https://github.com/oxidecomputer/omicron/pull/8925/commits/bbb204c6d821904ab933b31b9b204889851c1914